tools/bin/commands/git

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+git::description() {
+    echo "Git tools"
+}
+
+git::usage() {
+    cat <<EOT
+    has_app_module_changed <module>     prints true if the module has changed relative to master
+    touched_pl                          prints the list of /app modules uncommitted changes
+EOT
+}
+
+git::has_app_module_changed() {
+    modules="$@"
+
+    local changed=`\
+        git diff origin/master --numstat | cut -f 3 \
+        | grep "app/" | cut -c 5- \
+        | cut -d / -f 1 \
+        | awk 'NF' | sort | uniq \
+    `
+    for var in $changed; do
+        for m in $modules; do
+          if [ "$var" = "$m" ]; then
+              echo 'true'
+              return
+          fi
+        done
+    done
+    echo 'false'
+}
+
+git::touched_pl() {
+    cd `appdir` 
+    git status -s | cut -c 4- | grep / | grep -v '\.\.' | cut -d / -f 1 | awk 'NF' | sort | uniq | paste -s -d"," -
+}
+
+git::run() {
+
+    local command="$1"
+    shift
+
+    case $command in
+        has_app_module_changed) 
```

Comment:
Actually I would prefer the $(hasflag) and $(readopt) pattern as we use in the other commands. That would also easily allow for shortcuts.